crates/bin/prove_block/src/lib.rs

```diff
@@ -161,11 +164,20 @@ pub async fn prove_block(
     let blockifier_state_reader = AsyncRpcStateReader::new(provider_for_blockifier, BlockId::Number(block_number - 1));
 
     let mut blockifier_state = CachedState::new(blockifier_state_reader);
-    let tx_execution_infos = reexecute_transactions_with_blockifier(
-        &mut blockifier_state,
-        &block_context,
-        block_with_txs.transactions.iter().map(|tx| starknet_rs_to_blockifier(tx).unwrap()).collect(),
-    )?;
+
+    if block_with_txs.transactions.len() != traces.len() {
```

Comment:
I think this particular check could be an assert. It's basically supposed to be true unless the RPC does not respect the API contract.